task-service/openapi_directory.yaml

```diff
@@ -0,0 +1,132 @@
+          parameters:
+            - in: query
+              required: false
+              name: user
+              schema:
+                $ref: '#/components/schemas/user'
+              description: Filter directories by user
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/directory'
+        '400':
+          $ref: '#/components/responses/badRequest'
+    post:
+      summary: Create a new directory
+      description: Add a new directory
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/directory'
+      responses:
+        '201':
+          description: Directory created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/directory'
+  /directories/v1/{id}:
+    get:
+      summary: Get a directory by ID
+      description: Retrieve a specific directory by its ID
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Directory details
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/directory'
+        '400':
+          $ref: '#/components/responses/badRequest'
+    put:
+      summary: Update a directory
+      description: Update the details of an existing directory
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/directory'
+      responses:
+        '200':
+          description: Directory updated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/directory'
+        '400':
+          $ref: '#/components/responses/badRequest'
+    delete:
+      summary: Delete a directory
+      description: Remove a directory
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+      responses:
+        '204':
+          description: Directory deleted successfully
+        '400':
+          $ref: '#/components/responses/badRequest'
+components:
+  schemas:
+    directory:
+      id:
+        type: string
+      name:
+        type: string
+      createdDate:
+        type: string
+        format: date-time
+      updatedDate:
+        type: string
+        format: date-time
+      tasks:
+        type: array
+        items:
+          "$ref": "#/components/schemas/task"
+        description: The tasks in a directory
```

Comment:
we do not need this here. If the user wants to pull the tasks, they should call the tasks api